### PR TITLE
CB-8809 verify that AWS credentials' role is assumable without external Id

### PR DESCRIPTION
**Pull request by biharitomi**
See detailed description in the commit message.

### CODE REVIEW

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCredentialConnector.java [hunk 1]
```diff
@@ -168,6 +184,29 @@ private CloudCredentialStatus verifyAccessKeySecretKeyIsAssumable(CloudCredentia
         return credentialStatus;
     }
 
+    private void checkRoleIsAssumableWithoutExternalId(CredentialVerificationContext credentialVerificationContext, AwsCredentialView awsCredential) {
+        if (credentialVerificationContext.getCreationVerification()) {
+            String roleArn = awsCredential.getRoleArn();
+            try {
+                credentialClient.retrieveSessionCredentialsWithoutExternalId(awsCredential);
+                String message = String.format("CDP Control Pane is able to perform 'sts:AssumeRole' on '%s' role without external id. " +
```
Comment (doktoric) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCredentialConnector.java:192]: I am not sure but should we somehow show what is the external id which is missing ? I guess not but lets check

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCredentialConnector.java [hunk 2]
```diff
@@ -168,6 +184,29 @@ private CloudCredentialStatus verifyAccessKeySecretKeyIsAssumable(CloudCredentia
...
+                        "The role is vulnerable and could be exploited by confused deputy attacks. " +
+                        "Please update the role's trust relationship to have a condition on external id " +
+                        "or re-create the role with external id configured in the create role wizard.", roleArn);
+                LOGGER.warn(message);
+                throw new AwsPossibleConfusedDeputyException(message);
+            } catch (AmazonClientException ae) {
+                if (ae.getMessage().contains(ROLE_IS_NOT_ASSUMABLE_ERROR_MESSAGE_INDICATOR)) {
+                    String msg = String.format("Consider the specified role as secured. " +
+                            "CDP Control Pane is not authorized to perform sts:AssumeRole on '%s' role without external Id.", roleArn);
```
Comment (doktoric) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCredentialConnector.java:201]: I think here the problem is not the missing external id right ?

**Review by doktoric — CHANGES_REQUESTED** [review]
we should also mention this in mow-dev channel

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCredentialConnector.java (hunk 2 above)
Comment (biharitomi) [reply cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCredentialConnector.java:201]: That's a happy path and only a log is called to leave a sign about it.

**Review by doktoric — APPROVED** [review]

**Pull request merged**